Written by: mrobinson@webkit.org
Source: http://trac.webkit.org/changeset/112800
Upstream: yes
Reason: disable-geolocation breaks build

Index: Source/WebKit/gtk/webkit/webkitgeolocationpolicydecisionprivate.h
===================================================================
--- Source/WebKit/gtk/webkit/webkitgeolocationpolicydecisionprivate.h	(Revision 112279)
+++ Source/WebKit/gtk/webkit/webkitgeolocationpolicydecisionprivate.h	(Revision 112800)
@@ -25,14 +25,10 @@
 
 #include "webkitgeolocationpolicydecision.h"
 
-#if ENABLE(GEOLOCATION)
-
 extern "C" {
 
 WebKitGeolocationPolicyDecision* webkit_geolocation_policy_decision_new(WebKitWebFrame*, WebCore::Geolocation*);
 
 }
 
-#endif // ENABLE(GEOLOCATION)
-
 #endif
Index: Source/WebKit/gtk/webkit/webkitgeolocationpolicydecision.cpp
===================================================================
--- Source/WebKit/gtk/webkit/webkitgeolocationpolicydecision.cpp	(Revision 112279)
+++ Source/WebKit/gtk/webkit/webkitgeolocationpolicydecision.cpp	(Revision 112800)
@@ -53,18 +53,20 @@
     decision->priv = G_TYPE_INSTANCE_GET_PRIVATE(decision, WEBKIT_TYPE_GEOLOCATION_POLICY_DECISION, WebKitGeolocationPolicyDecisionPrivate);
 }
 
-#if ENABLE(GEOLOCATION)
 WebKitGeolocationPolicyDecision* webkit_geolocation_policy_decision_new(WebKitWebFrame* frame, WebCore::Geolocation* geolocation)
 {
-    g_return_val_if_fail(frame, NULL);
+#if ENABLE(GEOLOCATION)
+    g_return_val_if_fail(frame, 0);
     WebKitGeolocationPolicyDecision* decision = WEBKIT_GEOLOCATION_POLICY_DECISION(g_object_new(WEBKIT_TYPE_GEOLOCATION_POLICY_DECISION, NULL));
     WebKitGeolocationPolicyDecisionPrivate* priv = decision->priv;
 
     priv->frame = frame;
     priv->geolocation = geolocation;
     return decision;
+#else
+    return 0;
+#endif
 }
-#endif
 
 /**
  * webkit_geolocation_policy_allow
Index: Source/WebKit/gtk/ChangeLog
===================================================================
--- Source/WebKit/gtk/ChangeLog	(Revision 112279)
+++ Source/WebKit/gtk/ChangeLog	(Revision 112800)
@@ -1,3 +1,17 @@
+2012-03-31  Martin Robinson  <mrobinson@igalia.com>
+
+        [GTK] [Stable] --disable geolocation broken after recent merges
+        https://bugs.webkit.org/show_bug.cgi?id=82452
+
+        No review, as this is just a build fix.
+
+        Fix the geolocation build.
+
+        * webkit/webkitgeolocationpolicydecision.cpp:
+        (webkit_geolocation_policy_decision_new): When gelocation is off, just return null.
+        * webkit/webkitgeolocationpolicydecisionprivate.h: Activate webkit_geolocation_policy_decision_new
+        for non-Geolocation builds.
+
 2012-03-27  Martin Robinson  <mrobinson@igalia.com>
 
         Update the NEWS and version for the impending 1.8.0 release.
